api/src/test/java/com/epam/pipeline/manager/pipeline/RunScheduleManagerTest.java

```diff
@@ -103,214 +117,238 @@ public void setUp() {
         testPipeline.setRepository(TEST_REPOSITORY);
         testPipeline.setOwner(TEST_NAME);
         testPipeline.setRepositorySsh(TEST_PIPELINE_REPO_SSH);
-        pipelineDao.createPipeline(testPipeline);
+        testPipeline.setId(RUN_ID);
 
         createPipelineRun(RUN_ID, testPipeline.getId());
+
         runConfiguration = createRunConfiguration();
+        runConfiguration.setId(2L);
 
         PipelineUserVO userVO = new PipelineUserVO();
         userVO.setUserName(USER_OWNER);
-        userManager.createUser(userVO);
 
         testRunScheduleVO = getRunScheduleVO(RunScheduledAction.PAUSE, CRON_EXPRESSION1);
         testRunScheduleVO2 = getRunScheduleVO(RunScheduledAction.RESUME, CRON_EXPRESSION2);
         testRunScheduleVO3 = getRunScheduleVO(RunScheduledAction.RUN, CRON_EXPRESSION2);
 
+        MockitoAnnotations.initMocks(this);
+
+        mockRunConfigurationScheduleProvider =
+                Mockito.spy(new RunConfigurationScheduleProvider(mockMessageHelper, mockRunConfigurationManager));
+
+        doReturn(mockPipelineRunScheduleProvider)
+                .when(mockScheduleProviderManager).getProvider(ScheduleType.PIPELINE_RUN);
+        doReturn(mockRunConfigurationScheduleProvider)
+                .when(mockScheduleProviderManager).getProvider(ScheduleType.RUN_CONFIGURATION);
+        doReturn(mockPipelineUser).when(mockAuthManager).getCurrentUser();
     }
```

Comment:
The whole method can be greatly simplified by using existing `CreatorUtils` classes. Additionally it would be nice to replace existing private creator methods with the usage of existing `CreatorUtils` methods. Of course missing methods can be implemented.

```java 
private final RunConfiguration runConfiguration = ConfigurationCreatorUtils.getRunConfiguration(2L, TEST_NAME);
private final Pipeline testPipeline = PipelineCreatorUtils.getPipeline(RUN_ID, TEST_NAME);
private final PipelineRunScheduleVO testRunScheduleVO = getRunScheduleVO(RunScheduledAction.PAUSE, CRON_EXPRESSION1);
private final PipelineRunScheduleVO testRunScheduleVO2 = getRunScheduleVO(RunScheduledAction.RESUME, CRON_EXPRESSION2);
private final PipelineRunScheduleVO testRunScheduleVO3 = getRunScheduleVO(RunScheduledAction.RUN, CRON_EXPRESSION2);

@Before
public void setUp() {
    MockitoAnnotations.initMocks(this);
    doReturn(mockPipelineRunScheduleProvider)
            .when(mockScheduleProviderManager).getProvider(ScheduleType.PIPELINE_RUN);
    doReturn(mockRunConfigurationScheduleProvider)
            .when(mockScheduleProviderManager).getProvider(ScheduleType.RUN_CONFIGURATION);
    doReturn(mockPipelineUser).when(mockAuthManager).getCurrentUser();
}
```